### ISSUE TITLE
Wifi not working

### ISSUE THREAD

**Issue by bhardwahnitish19** [kfechter/LegionY530Ubuntu#97]
**Describe the bug**
Thanks for documenting al these steps so well. 

I am trying to install ubuntu 20.04 on my Legion laptop. Installation succeeds but I am stuck in enabling wifi.

I have gone through the troubleshooting section, but I get an error when I try to execute _sudo rmmod r8822be_

**rmmod: Error: Module r8822be is not currently loaded**

**To Reproduce**
echo "options r8822be aspm=0" | sudo tee /etc/modprobe.d/r8822be.conf
run the command sudo rmmod r8822be

**Expected behavior**
Wifi should detect the available networks

**Screenshots**

**System Info (please complete the following information):**
 - OS: Ubuntu 20.04
 - Kernel version Kernel: Linux 5.4.0-29-generic

I have tried a lot of things to get it working, but nothing worked. I would really appreciate your help.

**Comment by kfechter** [comment]
what is the output of the following command: lspci -v | grep -i wireless

and do you have a Legion Y530-15ICH or another model? it is possible that there are more models that aren't covered by ideapad_laptops "doesn't have a physical wifi switch" exclusion list.

try a sudo rmmod ideapad_laptop and see if the wifi and bluetooth power up.

**Comment by bhardwahnitish19** [comment]
Hi @kfechter 

Thanks for your quick response. Yes, I am using Legion Y530-15 INCH. Just ran the command:
`lspci -v | grep -i wireless`

> Got no output

`lspci | grep -i realtek`

Output: 
`nitish@Lenovo-Legion-Y530:~$ lspci | grep -i realtek
07:00.0 Network controller: Realtek Semiconductor Co., Ltd. RTL8822BE 802.11a/b/g/n/ac WiFi adapter (rev ff)
08:00.0 Ethernet controller: Realtek Semiconductor Co., Ltd. RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller (rev 15)
`

**Comment by bhardwahnitish19** [comment]
After executing this
`sudo tee /etc/modprobe.d/blacklist-ideapad.conf <<< "blacklist ideapad_laptop"`

Wifi is enabled but, it's not able to identify any available networks

**Comment by kfechter** [comment]
Ok, the ideapad_laptop module is ruled out as the issue, so the blacklist isn't needed. 

Could you try the following, you may need to do the copy commands as root using sudo. 

go to /lib/firmware/

here you will need to create a folder named rtw88

go to /lib/firmware/rtlwifi/

here you will need to copy the file rtl8822befw.bin

go to /lib/firmware/rtw88/

here you will need to insert the file you just copied (rtl8822befw.bin)

now you will need to rename the file to rtw8822b_fw.bin

reboot your system

Seems that kernel 5.2 may have broken wifi for the Realtek cards. I have an Intel card, so I can't test these steps.

**Comment by kfechter** [comment]
sudo mkdir /lib/firmware/rtw88
sudo cp /lib/firmware/rtlwifi/rtl8822befw.bin /lib/firmware/rtw88/rtw8822b_fw.bin

**Comment by bhardwahnitish19** [comment]
Hi, Thanks for these steps.

Just noticed that this file was already present at /lib/firmware/rtw88/rtw8822b_fw.bin

I executed the commands anyway and rebooted. Still no luck. Wi-Fi is still not able to detect any networks. :(

**Comment by bhardwahnitish19** [comment]
Just checked the kernel version:

**Kernel: Linux 5.4.0-29-generic**

**Comment by kfechter** [comment]
They must have broken something else.

Try







*git clone https://github.com/lwfinger/rtlwifi_new.git
<https://github.com/lwfinger/rtlwifi_new.git>cd rtlwifi_newgit checkout
extendedmakesudo make installsudo modprobe -r rtl8822sudo modprobe rtl8822*

*And then reboot.*

**Comment by bhardwahnitish19** [comment]
Hi,

I tried these commands, everything worked fine till

sudo modprobe -r rtl8822
sudo modprobe rtl8822

Getting an error:
```
nitish@Lenovo-Legion-Y530:~/rtlwifi_new$ sudo make install
make -C /lib/modules/5.4.0-29-generic/build M=/home/nitish/rtlwifi_new modules
make[1]: Entering directory '/usr/src/linux-headers-5.4.0-29-generic'
  Building modules, stage 2.
  MODPOST 19 modules
make[1]: Leaving directory '/usr/src/linux-headers-5.4.0-29-generic'
Making backups
Install rtlwifi SUCCESS
nitish@Lenovo-Legion-Y530:~/rtlwifi_new$ sudo modprobe -r rtl8822
modprobe: FATAL: Module rtl8822 not found.
nitish@Lenovo-Legion-Y530:~/rtlwifi_new$ sudo modprobe rtl8822
modprobe: FATAL: Module rtl8822 not found in directory /lib/modules/5.4.0-29-generic
```

**Comment by kfechter** [comment]
Yeah. It should be rtl8822be for those two commands. 

Also a reboot after running them.

**Comment by bhardwahnitish19** [comment]
After these steps and a reboot, can't see wi-fi itself. :(

**Comment by kfechter** [comment]
Ugh. They must have really broken the Realtek cards with 5.2.

In the rtlwifi_new directory. Do a sudo make uninstall to revert back to
the older module. Its powering up the radios, so something else is going
on.

Unfortunately I do not have a realtek wifi card for testing, so I can only
provide the information I find online.

**Comment by kfechter** [comment]
looks like there may already be open bugs
https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1853050
https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1872984

someone with a Yoga replied that a BIOS update fixed it, however, the latest BIOS for the legion is 8JCN53WW from December 2019.

**Comment by RobertRozic** [comment]
Can confirm that installing latest BIOS fixed my wifi problems on 20.04.
I have RTL8822BE wifi card.

**Comment by bhardwahnitish19** [comment]
@RobertRozic  That's a great news, how did you installed the latest BIOS?

**Comment by RobertRozic** [comment]
@bhardwahnitish19 I have dual boot, so installed it from windows. Sadly, it is isn't performing well and it is losing connection very often. I think i will order Intel 9560NGW wireless card so i can avoid this problems. I will report my experiences here.

**Comment by bhardwahnitish19** [comment]
Ohh, sadly, I don't have dual boot to try this.

**Comment by RobertRozic** [comment]
I have also had troubles with wifi on windows as well. 
I have been using 19.04 for about year without problems but now with fresh installed 20.04, and bios updated it is behaving similar as on windows. I was reading some other threads and best advice is to replace wifi card with Intel one.

**Comment by RobertRozic** [comment]
Update
After many tries to fix drivers for RTL8822BE i finally decided to order Intel 9560 Wireless card.
I regret not doing this earlier, since that could have saved me many hours.
Seems that RTL8822BE drivers are broken after 5.2 kernel.
I suggest buying that wifi card, it is around $10.

**Comment by kfechter** [comment]
9560 has worked well in my experience. Not sure what the breakdown is, but seems like a lot of models are coming with Realtek cards. My model came with with the 9560, but was also an earlier review unit, so maybe they switched wifi cards for newer production.

**Comment by logg926** [comment]
> Can confirm that installing latest BIOS fixed my wifi problems on 20.04.
> I have RTL8822BE wifi card.

I am still having this issue. May I know what was your BIOS version number?
